argus/action_manager/windows.py

```diff
@@ -115,32 +116,76 @@ def get_installation_script(self):
+
+        check_cmd = r'& "{}\python.exe" -c "import cloudbaseinit"'.format(
+            python_path)
+
+        try:
+            self._client.run_remote_cmd(
+                cmd=check_cmd, command_type=util.POWERSHELL)
+        except exceptions.ArgusError as exc:
+            LOG.error("Cloudbase-Init installation failed: %s", exc)
+            return False
+
+        LOG.info("Cloudbase-Init was successfully installed!")
+        return True
+
+    def cbinit_cleanup(self, cbinit_path):
+        """Clean up Cloudbase-Init."""
+        LOG.info("Cleaning up Cloudbase-Init ...")
+
+        self.rmdir("{}\\..".format(cbinit_path))
+
     def install_cbinit(self):
-        """Run the installation script for CloudbaseInit."""
-        LOG.debug("Installing Cloudbase-Init ...")
+        """Run the installation script for Cloudbase-Init."""
+        LOG.debug("Trying to install Cloudbase-Init ...")
 
         installer = "CloudbaseInitSetup_{build}_{arch}.msi".format(
             build=self._conf.argus.build,
             arch=self._conf.argus.arch
         )
         # TODO(cpopa): the service type is specific to each scenario,
         # find a way to pass it
-        LOG.info("Run the downloaded installation script "
-                 "using the installer %r.", installer)
 
-        parameters = '-installer {}'.format(installer)
-        try:
-            self.execute_powershell_resource_script(
-                resource_location='windows/installCBinit.ps1',
-                parameters=parameters)
-        except exceptions.ArgusError:
-            # This can happen for multiple reasons,
-            # but one of them is the fact that the installer
-            # can't be installed through WinRM on some OSes
-            # for whatever reason. In this case, we're falling back
-            # to use a scheduled task.
-            LOG.debug("Cannot install, deploying using a scheduled task.")
-            self._deploy_using_scheduled_task(installer)
+        for _ in range(util.RETRY_COUNT):
+            LOG.info("Running the downloaded installation script "
+                     "using the installer %r.", installer)
+
+            parameters = '-installer {}'.format(installer)
+            try:
+                self.execute_powershell_resource_script(
+                    resource_location='windows/installCBinit.ps1',
+                    parameters=parameters)
+            except exceptions.ArgusError:
+                # This can happen for multiple reasons,
+                # but one of them is the fact that the installer
+                # can't be installed through WinRM on some OSes
+                # for whatever reason. In this case, we're falling back
+                # to use a scheduled task.
+                LOG.debug("Could not install Cloudbase-Init!")
+
+                LOG.info("Deploying Cloudbase-Init using a scheduled task ...")
+                self._deploy_using_scheduled_task(installer)
```

Comment:
In previous implementation, this case was left untreated like a Plan B.
Should I include this method in the new `install_cbinit` flow?